netcore/System.Private.CoreLib/src/System.Runtime.Loader/AssemblyLoadContext.cs

```diff
@@ -130,6 +131,52 @@ internal static Assembly DoAssemblyResolve (string name)
 			return ResolveSatelliteAssembly (gchALC, new AssemblyName (assemblyName));
 		}
 
+		private static void MonoResolveUnmanagedDll(string unmanagedDllName, IntPtr gchManagedAssemblyLoadContext, ref IntPtr dll)
+		{
+			AssemblyLoadContext context;
+			// This check exists because the function can be called early in startup, before the default ALC is initialized
+			if (gchManagedAssemblyLoadContext == IntPtr.Zero)
+				context = AssemblyLoadContext.Default;
+			else
+				context = (AssemblyLoadContext)(GCHandle.FromIntPtr(gchManagedAssemblyLoadContext).Target)!;
+			dll = context.LoadUnmanagedDll(unmanagedDllName);
+		}
+
+		private static void MonoResolveUnmanagedDllUsingEvent(string unmanagedDllName, Assembly assembly, IntPtr gchManagedAssemblyLoadContext, ref IntPtr dll)
+		{
+			AssemblyLoadContext context;
+			// This check exists because the function can be called early in startup, before the default ALC is initialized
+			if (gchManagedAssemblyLoadContext == IntPtr.Zero)
+				context = AssemblyLoadContext.Default;
+			else
+				context = (AssemblyLoadContext)(GCHandle.FromIntPtr(gchManagedAssemblyLoadContext).Target)!;
```

Comment:
Seems like this could be extracted to `AssemblyLoadContext  GetAssemblyLoadContext (IntPtr gchManagedAssemblyLoadContext)`

Reply:
Will do.